list/letters-filter/index.js

```diff
@@ -0,0 +1,153 @@
+/**@jsx*/
+let builder = require('focus').component.builder;
+let stylable = require('../../mixin/stylable');
+let type = require('focus').component.types;
+
+let alphabetLetters = 'abcdefghijklmnopqrstuvwxyz';
+
+let lettersFilterMixin = {
+    mixins: [stylable],
+    /**
+     * Component display name.
+     */
+    displayName: 'list-letters-filter',
+
+    /**
+     * Letters.
+     * @type {Array}
+     */
+    letters: alphabetLetters.toUpperCase().split(''),
+
+    /** @inheritedDoc */
+    getDefaultProps() {
+        return {
+            /**
+             * Current selected letter.
+             * @type {string}
+             */
+            selectedLetter: '',
+
+            /**
+             * Available letters.
+             * @type {string}
+             */
+            availableLetters: '',
+
+            /**
+             * Handle letter selection action (must be overridden by calling component).
+             * @param letter
+             */
+            handleLetterSelection: function(letter) { console.warn('selected letter: ', letter); } // Action on selection click
+        };
+    },
+
+    /** @inheritdoc */
+    propTypes: {
+        selectedLetter: type('string'),
+        availableLetters: type('string')
+    },
+
+    /** @inheritdoc */
+    getInitialState(){
+        return{
+            selectedLetter: this.props.selectedLetter
+        }
+    },
+
+    /** @inheritdoc */
+    componentWillReceiveProps(newProps){
+        this.setState(
+            {
+                selectedLetter: newProps.selectedLetter,
+                availableLetters: newProps.availableLetters
+            }
+        );
+    },
+
+    /**
+     * Handle letter selection.
+     * @private
+     */
+    _selectionFunction(event) {
+        var letter = event.target.getAttribute('data-letter');
+        if(this._isLetterAvailable(letter)){
+            this.setState({selectedLetter: letter});
```

Comment:
done. I agree with you. it's better to use callback function of setState() method.